### PR TITLE
fix!: Update case syntax to use brackets over braces

### LINKED ISSUE #3462: `case` should use `[]` instead of `{}`
### What's up?

Currently, this is syntax for case:

```prql
from employees
derive distance = case {
  city == "Calgary" => 0,
  city == "Edmonton" => 300,
}
```

In the last dev call, @snth made a point that `case` syntax should be more similar to arrays than tuples, since the elements of the case block:
- must have the same type,
- are not named.

This is the proposed syntax:

```prql
from employees
derive distance = case [
  city == "Calgary" => 0,
  city == "Edmonton" => 300,
]
```

### PR DESCRIPTION
**Pull request by AaronMoat**
BREAKING CHANGE: This is a breaking change to case syntax.

Resolves #3462

### CODE REVIEW

**Comment by eitsupi** [comment]
Hi, thanks for working on this!
We can't merge this into the current version (See #3477), so I'll turn it into a draft to prevent merging.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 **Language**:
 
+- _Breaking:_ Case syntax now uses brackets `[]` rather than braces `{}`.
+
+  To convert previous PRQL queries to this new syntax simply change
+  `case { ... }` to `case [ ... ]`.
+
```
Comment (eitsupi) [inline CHANGELOG.md]: This should move to 0.10.0.

**Review by eitsupi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (AaronMoat) [reply CHANGELOG.md]: I've moved to a new unreleased section, though will probably get conflicts that need fixing when #3477 is in.

**Comment by max-sixty** [comment]
Great work! Thanks @AaronMoat .

Let's merge as soon as we release 0.9.5?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,32 @@
...
+**Features**:
+
+**Fixes**:
+
+- Update case syntax to brackets `case [ ... ]` over braces `case { ... }`
+  (@AaronMoat, #3517)
```
Comment (max-sixty) [inline CHANGELOG.md]: I think we can consolidate this into the changelog above, so we only have one entry?

**Comment by max-sixty** [comment]
We can merge now!

**Review by eitsupi — APPROVED** [review]
Thanks!

**Pull request merged**